src/udev/udev-worker.c

```diff
@@ -206,20 +206,24 @@ static int worker_process_device(UdevWorker *worker, sd_device *dev) {
                 /* in case rtnl was initialized */
                 worker->rtnl = sd_netlink_ref(udev_event->rtnl);
 
-        if (udev_event->inotify_watch) {
-                r = udev_watch_begin(worker->inotify_fd, dev);
-                if (r < 0 && r != -ENOENT) /* The device may be already removed, ignore -ENOENT. */
-                        log_device_warning_errno(dev, r, "Failed to add inotify watch, ignoring: %m");
-        }
+        if (!device_for_action(dev, SD_DEVICE_REMOVE)) {
```

Comment:
hmm, a similar conditionalization should probably cover the `worker_lock_whole_disk()` and `blockdev_read_only` pars above, no? They don't really make much sense when a device is going away, we are not going to probe its superblock then, after all, nor are going to be able to make it read-only...